build-logic/binary-compatibility/src/main/kotlin/gradlebuild/binarycompatibility/sources/KotlinSourceQueries.kt

```diff
@@ -240,9 +256,18 @@ val CtClass.isKotlinFileFacadeClass: Boolean
 
 private
 fun CtBehavior.firstParameterMatches(ktTypeReference: KtTypeReference): Boolean =
-    parameterTypes.isNotEmpty() && (primitiveTypeStrings[parameterTypes.first().name] ?: parameterTypes.first().name).endsWith(ktTypeReference.text)
+    parameterTypes.firstOrNull()?.isLikelyEquivalentTo(ktTypeReference) ?: false
 
 
+private
+fun CtClass.isLikelyEquivalentTo(ktTypeReference: KtTypeReference): Boolean {
+    if (ktTypeReference.text.contains(" -> ")) {
```

Comment:
❓ is it checking the string representation for an arrow with spaces around it here the best we can do? It feels fragile.

Reply:
Yes, I'm interested in better solutions here. It might be more appropriate to strip generics first, but that also isn't the best option. I'd prefer to fully introspect, but I'm not familiar with Kotlin PSI.